int8 map_id
int8 linkpoint_id
